src/Sdk/DTWebApi/WebApi/TimelineRecord.cs

```diff
@@ -299,13 +310,53 @@ public TimelineRecord Clone()
             return new TimelineRecord(this);
         }
 
-        [DataMember(Name = "Issues", EmitDefaultValue = false, Order = 60)]
+        [OnDeserialized]
+        private void OnDeserialized(StreamingContext context)
+        {
+            this.EnsureInitialized();
+        }
+
+
+        /// <summary>
+        /// DataContractSerializer bypasses all constructor logic and inline initialization!
+        /// This method takes the place of a workhorse constructor for baseline initialization.
+        /// The expectation is for this logic to be accessible to constructors and also to the OnDeserialized helper.
+        /// </summary>
+        private void EnsureInitialized()
+        {
+            // Note that ?? is a short-circuiting operator.  (??= would be preferable, but it's not supported in the .NET Framework version currently used by actions/runner.)
+
+            // De-nullify the following historically-nullable ints.
+            // (After several weeks in production, it may be possible to eliminate these nullable backing fields.)
+            m_errorCount = m_errorCount ?? 0;
+            m_warningCount = m_warningCount ?? 0;
+            m_noticeCount = m_noticeCount ?? 0;
+
+            m_issues = m_issues ?? new List<Issue>();
+            m_previousAttempts = m_previousAttempts ?? new List<TimelineAttempt>();
+            this.Attempt = Math.Max(this.Attempt, 1);
+
+            // Ensure whatever content may have been deserialized for m_values is backed by a case-insensitive Dictionary.
+            var empty = Enumerable.Empty<KeyValuePair<string, VariableValue>>();
+            m_variables = new Dictionary<string, VariableValue>(m_variables ?? empty, StringComparer.OrdinalIgnoreCase);
```

Comment:
`DataContractSerializer` has no way of knowing that what we really want is a case-insensitive Dictionary.  We have to take whatever data may have been deserialized and re-bundle it into a new (case-insensitive) Dictionary.